services/agents/versioner.go

```diff
@@ -0,0 +1,137 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package agents
+
+import (
+	"github.com/percona/pmm/api/agentpb"
+	"github.com/pkg/errors"
+)
+
+// VersionerService provides methods for retrieving versions of different software.
+type VersionerService struct {
+	r *Registry
+}
+
+// NewVersionerService returns new versioner service.
+func NewVersionerService(registry *Registry) *VersionerService {
+	return &VersionerService{
+		r: registry,
+	}
+}
+
+// Version contains version and error fields.
+// Both Version and Error fields could be empty if requested software is not installed.
+type Version struct {
+	Version string
+	Error   string
+}
+
+// Software interface.
+type Software interface {
+	isSoftware()
+}
+
+// Mysqld represents mysqld software.
+type Mysqld struct{}
+
+func (*Mysqld) isSoftware() {}
```

Comment:
Why do we need this as types, why not just string constants?

Reply:
later for the mongodb I will need couple of fields for requesting db version.

Reply:
Ah, got it!